cpp/src/CMakeLists.txt

```diff
@@ -129,6 +128,234 @@ if(USE_AVX512)
   endif ()
 endif()
 
+# Build Arrow macro
+macro(build_arrow STATIC_ARROW)
+  set(ARROW_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep-install")
+  message(STATUS "ARROW_PREFIX: ${ARROW_PREFIX}")
+  set(ARROW_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep")
+  message(STATUS "ARROW_SOURCE_DIR: ${ARROW_SOURCE_DIR}")
+  set(ARROW_INCLUDE_DIR "${ARROW_PREFIX}/include")
+  set(BINARY_RELEASE_DIR "${root_directory}/releases")
+
+  # Load libarrow
+  set(ARROW_LIB_NAME arrow)
+
+  # Load libgandiva
+  set(GANDIVA_LIB_NAME gandiva)
+
+  # Load libparquet
+  set(PARQUET_LIB_NAME parquet)
+
+  # Set up arrow cmake args
+  set(MAKE_BUILD_ARGS
+      -DCMAKE_INSTALL_PREFIX:PATH=${ARROW_PREFIX}
+      -DARROW_BUILD_STATIC=ON
+      -DARROW_BUILD_SHARED=ON
+      -DARROW_DEPENDENCY_SOURCE=BUNDLED
+      -DARROW_COMPUTE=ON
+      -DARROW_GANDIVA_JAVA=ON
+      -DARROW_GANDIVA=ON
+      -DARROW_PARQUET=ON
+      -DARROW_HDFS=ON
+      -DARROW_BOOST_USE_SHARED=OFF
+      -DARROW_JNI=ON
+      -DARROW_DATASET=ON
+      -DARROW_WITH_PROTOBUF=ON
+      -DARROW_WITH_SNAPPY=ON
+      -DARROW_WITH_LZ4=ON
+      -DARROW_WITH_ZSTD=OFF
+      -DARROW_WITH_BROTLI=OFF
+      -DARROW_WITH_ZLIB=OFF
+      -DARROW_FILESYSTEM=ON
+      -DARROW_JSON=ON
+      -DARROW_FLIGHT=OFF
```

Comment:
let's add this missing param, "-DARROW_WITH_FASTPFOR=ON"

Reply:
Done to turn it on in 2nd commit.